hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/TokenAirdropValidator.java

```diff
@@ -118,6 +117,12 @@ public void validateSemantics(
                 // 1. Validate token associations
                 validateFungibleTransfers(
                         context.payer(), senderAccount, tokenId, senderAccountAmount.get(), tokenRelStore);
+                totalFungibleTransfers += xfers.transfers().size();
+
+                // Verify that the current total number of (counted) fungible transfers does not exceed the limit
+                validateTrue(
```

Comment:
Can we have a `//FUTURE` comment or better an issue in this repo to refactor this and `CryptoTransferValidator.validateSemantics` so that they use common code for these common checks.?

(There are differences in the specific validations done for crypto xfer vs token airdrop, but these _structural_ checks should be common code.  IMO.)

Reply:
Agreed. Since this is a release-blocking issue I suspect we should not refactor this here, but I'd also like to see the common validations extracted

Reply:
I agree with both of you.
I will create the following issue for our happy Thursday.